### ISSUE TITLE
pdf.js on windows phone

### ISSUE THREAD

**Issue by tfierens** [mozilla/pdf.js#6535]
Hi,

We're writing an app with Xamarin.Forms and we're trying to get pdf.js to work with iOs, Android and Windows Phone. So far we've managed to get it to work with Android but can't figure out how to use it with Windows Phone.

In Android, we can open the file by creating the following in uri:
 
PdfWebViewer.Source = (new Uri("file:///android_asset/pdfjs/web/viewer.html?file=file:///data/data/AdestAPMobile.Droid/files/Temp/TempPdf.pdf"));

In Windows Phone (JavaScript is enabled), it will display the viewer when we have:

PdfWebViewer.Source = new Uri("/Assets/pdfjs/web/viewer.html");

but whenever we add file=... it displays a page not found.

The path where the file is saved is:

C:\\Data\\Users\\DefApps\\AppData\\{3F3D90EF-3BA3-411C-961F-D1F9064CDD05}\\local\\Temp\\TempPdf.pdf

I've tried opening in a similar way as Android, but to no avail:

PdfWebViewer.Source = (new Uri("/Assets/pdfjs/web/viewer.html/file=file:///Data/Users/DefApps/AppData/{3F3D90EF-3BA3-411C-961F-D1F9064CDD05}/local/Temp/TempPdf.pdf"));

I've tried lots of other combinations but still can't get it to work.

Does anyone have any idea?

Thanks.

Thierry

**Comment by yurydelendik** [comment]
Can you load this file using regular XHR?  (PDF.js is using browser's XHR from the worker to load PDF)

**Comment by tfierens** [comment]
Hi,

Thanks for getting back to me. I'll go research this further and revert back when I have something more concrete but I'm not familiar with JavaScript, so I'll probably have to figure this part out. Thanks.

**Comment by timvandermeij** [comment]
This sounds more like an implementation issue than a bug in PDF.js. Since this issue tracker is primarily used for tracking bugs in PDF.js, I'm closing this. If there are further questions, contact us on IRC or the mailing list.

**Comment by tfierens** [comment]
Hi Tim,

Apologies if I used the wrong place to log my issue and thanks for closing it.

Take care,

Thierry

Sent from my Windows Phone
________________________________

**Comment by sivashankararumugam** [comment]
@yurydelendik, using Xamarin forms I need to display pdf file in window phone 10  & 8.1 (UWP) 

My pdf file is in below path "C:\Users\ramanans\AppData\Local\Packages\698343db-d99e-4d49-b082-6c429844e50d_y1rwc9yf20874\LocalState\prajesh.pdf"

I have tried with many options
string savedpath = WebUtility.UrlEncode(ApplicationData.Current.LocalFolder.Path + "/" + "Prajesh.pdf");
Control.Source = new Uri("ms-appx-web:///Assets/pdfjs/web/viewer.html?file=" + savedpath);

**Comment by maurelio1979** [comment]
Hi!

I managed to work this out by saving the file in Assets folder (under a new folder named Content I created) and then accessing it using Windows.ApplicationModel.Package.Current.InstalledLocation (as per https://blogs.msdn.microsoft.com/metroapps/2012/07/15/access-your-application-assets-folder/ )

Below is the code snippet which saves the file:

`StorageFolder localFolder = Windows.ApplicationModel.Package.Current.InstalledLocation;
            IStorageFile storageFile = await localFolder.CreateFileAsync(@"Assets\Content\" + filename,
                                                    CreationCollisionOption.ReplaceExisting);
            await FileIO.WriteBytesAsync(storageFile, bytes);`


... and the snippet which inserts it into WebView using pdf.js:

`Control.Source = new Uri(string.Format("ms-appx-web:///Assets/pdfjs/web/viewer.html?file={0}",
                        string.Format("ms-appx-web:///Assets/Content/{0}", WebUtility.UrlEncode(filename))));`

... where filename is something like "file.pdf" (without path).

I hope this helps!
